client/src/BidList.js

```diff
@@ -0,0 +1,143 @@
+	const [loading, setLoading] = useState(true);
+	const navigate = useNavigate();
+	const [pagination, setPagination] = useState({
+		itemsPerPage: 10,
+		currentPage: currentPage,
+		totalPages: 1,
+	});
+
+	const fetchBids = async (tenderId, page) => {
+		setLoading(true);
+		try {
+			const data = await get(`/api/bid?tender_id=${tenderId}&page=${page}`);
+			setBid(data.results);
+			setPagination(data.pagination);
+			setErrorMsg(null);
+		} catch (error) {
+			setErrorMsg("Error fetching bids!");
+		} finally {
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		fetchBids(tenderId, currentPage);
+	}, [currentPage, tenderId]);
+
+	const handleBidStatusChange = async (bidId, status) => {
+		const response = await post(`/api/bid/${bidId}/status`, { status });
+		if (response.code === "SUCCESS") {
+			setUpdatedStatus({ id: bidId, status: status });
+		} else {
+			setErrorMsg("Error fetching bids!");
+		}
+	};
+
+	if (bid.length === 0) {
+		return <div className="msg">No Bid Submited yet!!</div>;
+	}
+
+	const loadNextPage = () => {
+		if (pagination.currentPage < pagination.totalPages && !loading) {
+			navigate(`/bidding/page/${pagination.currentPage + 1}`);
+		}
+	};
+
+	const loadPreviousPage = () => {
+		if (pagination.currentPage > 1 && !loading) {
+			navigate(`/bidding/page/${pagination.currentPage - 1}`);
+		}
+	};
+
+	if (errorMsg !== null) {
+		return <div>{errorMsg}</div>;
+	}
+
+	if (loading) {
+		return <div>Loading!!</div>;
+	}
+
+	return (
+		<main>
+			<h2 className="heading">Bids</h2>
+			<div className="container">
+				{errorMsg && <p className="error-message">{errorMsg}</p>}
+				{bid.map((bid) => (
+					<div className="card" key={bid.bid_id}>
+						<p className="posted-on">
+							Submited on
+							<span className="posted-on-date">
+								{new Date(bid.bidding_date).toLocaleDateString()}
+							</span>
+							<span className="bid-status">
+								{updateStatus.id === bid.bid_id
+									? updateStatus.status
+									: bid.status}
+							</span>
+						</p>
+						<p className="title">
+							<strong>Bidder Id:</strong> {bid.bidder_id} |
+							<strong>Bidder Name: </strong>
+							{bid.first_name + " " + bid.last_name} |
+							<strong>Proposed Project Duration: </strong>
+							{bid.suggested_duration_days} days
+						</p>
+						<h4>Cover letter:</h4>
+						<p className="cover-letter"> {bid.cover_letter}</p>
+						<div className="btn-container">
+							{bid.attachment && (
+								<button className="btn">
+									<a href={bid.attachment} download>
+										Download Attachments
+									</a>
+								</button>
+							)}
+							<button
+								className="btn"
+								onClick={() => handleBidStatusChange(bid.bid_id, "Awarded")}
+							>
+								Accept
+							</button>
+							<button
+								className="btn"
+								onClick={() => handleBidStatusChange(bid.bid_id, "Rejected")}
+							>
+								Reject
+							</button>
+						</div>
```

Comment:
These buttons should be shown conditionally. If the bid is already accepted, there should be no option to accept. The same for rejected

Reply:
So, once buyer accepted or rejected a bid, both buttons will not be shown? and other bids button will be shown. correct?

Reply:
Yes, I think that would be fine. If a bid is `accepted` or `rejected` no buttons should be shown. Since the backend rejects all other bids when a bid is accepted, it might be a good idea to refetch the current page when a bid is accepted.

Reply:
So, only all button will be disappeared when buyer accept a bid. But when buyer click on reject, no button will be disappeared?

Reply:
Then I think we don't need `rejec`t button. correct?

Reply:
The buyer could decide to `reject` a specific bid before choosing which one to accept, so I think it would still be useful to have it

Reply:
So, when buyer reject a bid, Its mean buttons will be shown. But you are saying that even if `rejected` no button should be shown.

> Yes, I think that would be fine. If a bid is `accepted` or `rejected` no buttons should be shown. Since the backend rejects all other bids when a bid is accepted, it might be a good idea to refetch the current page when a bid is accepted.

Reply:
If a bid is `accepted` or `rejected` no buttons should be shown.

If a buyer rejects a bid, then that bid will become `rejected` so buttons won't be shown any more.